sisl/_blender.py

```diff
@@ -0,0 +1,501 @@
+            BlenderScene.data respectively. *You shouldn't need any blender imports inside your function*
+        '''
+
+        #Define the output path
+        self.context.scene.render.filepath = filepath or self.filepath
+
+        #Define the renderer
+        self.context.scene.render.engine = engine
+
+        image_settings = {"file_format": file_format, **image_settings}
+
+        for key, val in image_settings.items():
+            setattr(self.context.scene.render.image_settings, key, val)
+
+        bpy.context.scene.render.film_transparent = transparent_bg
+        bpy.data.worlds["World"].node_tree.nodes["Background"].inputs[0].default_value = bg_color
+        bpy.data.worlds["World"].node_tree.nodes["Background"].inputs[1].default_value = bg_strength
+
+
+        if callable(before_render):
+            before_render(self)
+
+        #And render :)
+        self.ops.render.render('INVOKE_DEFAULT', write_still=True)
+
+if BLENDER_AVAIL:
+    def blender(obj, scene=None, lighting=True, lighting_settings={}, camera=True, camera_settings={}, render=True, render_settings={}):
+        '''
+
+        Creates a blender scene containing the given object and renders it if desired.
+
+        Parameters
+        ----------
+        obj: any sisl object with a __blender__() method
+            The object to be added to the 3D view.
+        scene: BlenderScene, optional
+            The scene in which to put the 3D representation of the object. If not provided, a new scene will be created.
+        lighting: bool, optional
+            Whether lighting should be set up or not.
+        lighting_settings: dict, optional
+            Dict whose values will be passed directly to `BlenderScene.lighting()`
+
+            See the `BlenderScene.lighting` method to know what can be tuned.
+        camera: bool, optional
+            Whether the camera should be set up or not.
+        camera_settings: dict, optional
+            Dict whose values will be passed directly to `BlenderScene.camera()`
+
+            See the `BlenderScene.camera` method to know what can be tuned.
+        render: bool, optional
+            Whether the scene should be rendered or not
+        render_settings: dict, optional
+            Dict whose values will be passed directly to `BlenderScene.render()`
+
+            See the `BlenderScene.render` method to know what can be tuned.
+        
+        
+        '''
+        if scene is None: 
+            scene = BlenderScene()
+            
+        scene.add(obj)
+
+        if lighting:
+            scene.lighting(**lighting_settings)
+
+        if camera:
+            scene.camera(**camera_settings)
+
+        if render:
+            scene.render(**render_settings)
+
+        return scene
+else:
+    def blender(obj, *args, **kwargs):
+        raise BLENDER_IMPORT_EXCEPTION
+
+# Clean up
+del BLENDER_AVAIL
+del INSIDE_BLENDER
+del BLENDER_ASPACKAGE
+
+if __name__ == "__main__":
+    
```

Comment:
This should go in a file in a test directory to disconnect it from the actual code.

Reply:
I forgot to remove it, I wasn't using it anymore anyway.

How could I implement tests for this file? It needs the right environment to run and this will only happen when the methods are being run inside blender.